src/shared/Core.Tests/Trace2Tests.cs

```diff
@@ -30,4 +56,112 @@ public void TryGetPipeName_Windows_Returns_Expected_Value(string input, string e
         Assert.True(isSuccessful);
         Assert.Matches(actual, expected);
     }
+
+    [Theory]
+    [InlineData("", 0)]
+    [InlineData("foo", 0)]
+    [InlineData("foo/bar", 1)]
+    [InlineData("foo/bar/baz", 2)]
+    public void TryGetProcessDepth_Returns_Expected_Depth(string input, int expected)
+    {
+        SidManager.Sid = input;
+        var commandContext = new TestCommandContext();
+        var trace2 = new Trace2(commandContext);
+        trace2.Initialize(DateTimeOffset.UtcNow);
+
+        var actual = trace2.GetProcessDepth();
+
+        Assert.Equal(expected, actual);
+    }
+
+    [Fact]
+    public void BuildSpan_NoMatch_Returns_Expected_String()
+    {
+        var expected = "|          ";
+        var component = new PerformanceFormatComponent()
+        {
+            Name = "Foo",
+            SpanSize = 10,
+            BeginPadding = 1,
+            EndPadding = 1
+        };
+
+        var trace2Message = new TestTrace2Message();
+        var actual = trace2Message.BuildSpan(component);
+
+        Assert.Equal(expected, actual);
+    }
+
+    [Theory]
+    [InlineData(0.013772,    "|  0.013772 ")]
+    [InlineData(26.316083,   "| 26.316083 ")]
+    [InlineData(100.316083,  "|100.316083 ")]
+    public void BuildSpan_Time_Match_Returns_Expected_String(double input, string expected)
+    {
+        var component = new PerformanceFormatComponent()
+        {
+            Name = "ElapsedTime",
+            SpanSize = 11,
+            BeginPadding = 2,
+            EndPadding = 1
+        };
+
+        var trace2Message = new TestTrace2Message()
+        {
+            ElapsedTime = input
+        };
+
+        var actual = trace2Message.BuildSpan(component);
+
+        Assert.Equal(expected, actual);
```

Comment:
Exact matching the string formatting of a float/double is iffy because of round-off and other machine precision issues.  This seems to work at this precision level, but would probably fail to give the right answer if you tried "10000....000.123456" because the mantissa doesn't have enough resolution.  I'm not saying we should change this, but just be aware that testing floats is tricky.

Mabye run the result thru sscanf("|%g ",&v) and see it if is within an epsilon of the input value.  But this may be overkill.

Reply:
With Xunit we can use the `Assert.Equals(double, double, int)` overload to specify a precision for comparing the floating point numbers:

<img width="685" alt="image" src="https://user-images.githubusercontent.com/5658207/225146649-db752eba-85b7-498e-b4bf-86d222e02967.png">

Reply:
Going to leave as-is for now. Point is completely valid, but I feel the test is useful as it stands for checking the alignment of values we'll see in the majority of cases.